app/Models/Device.php

```diff
@@ -270,6 +270,14 @@ public function formatUptime($short = false)
         return Time::formatInterval($this->uptime, $short);
     }
 
+    public function formatDowntime($short = false)
+    {
+        $ret = static::where('hostname', $this->hostname)->first(['hostname', 'last_polled']);
```

Comment:
Why are you running an sql query here?
`$this->last_polled` holds the value.

Reply:
Why a seperate function from formatUptime?

Reply:
> Why are you running an sql query here?
> `$this->last_polled` holds the value.
$this->last_polled returns null. I don't know why. Same with some other keys (last_ping...)

Reply:
Device->formatUptime($short = false) does not take any time input argument. I cannot reuse it.

Reply:
if last_polled holds null, it is because there was a select statement done excluding that value when the device model was loaded.

Reply:
Device->formatUpDownTime($short = false) {
time = down ? now - last_polled : uptime
return format(uptime)

Reply:
also, what do you think about putting a minus (-) sign before all downtimes?  that way you know what it is even out of context.

Reply:
and i'd like it in a red font color ... so it also differs from normal uptime text.
what do you think about it @murrant

Reply:
> if last_polled holds null, it is because there was a select statement done excluding that value when the device model was loaded. (or the device wasn't polled yet)

I will try to search for that statement

Reply:
> Device->formatUpDownTime($short = false) {
> time = down ? now - last_polled : uptime
> return format(uptime)

OK for that

Reply:
> and i'd like it in a red font color ... so it also differs from normal uptime text.
> what do you think about it @murrant

@SourceDoctor please tell me how to do it with href title !

Reply:
you'd have to use a css popup (probably via bootstrap)

Reply:
I've added downtime suffix
![image](https://user-images.githubusercontent.com/47607835/83739995-937c0c00-a656-11ea-82fe-87b002d7bef0.png)

Reply:
formatUpDownTime function factorization done